From 7778e3d31162ab6a11b502f6b945767b4b8c65f9 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 26 Sep 2022 19:51:31 +0000
Subject: [PATCH] Fix build with Qt 6.4

* fix:
git/adaptors/alsadaptor/alsadaptor.cpp:203:30: error: use of deleted function 'QVariant::QVariant(T) [with T = char*; typename std::enable_if<disjunction_v<std::is_pointer<_Tp>, std::is_member_pointer<_Tp> >, bool>::type <anonymo
us> = false]'

and

../../../git/examples/filterplugin/samplefilter.h:43:45: error: invalid use of incomplete type 'class QObject'
../../../git/examples/filterplugin/samplefilter.h:43:45: error: invalid use of incomplete type 'class QObject'
moc_samplefilter.cpp:60:43: error: incomplete type 'QObject' used in nested name specifier
moc_samplefilter.cpp:60:60: error: no matching function for call to 'QMetaObject::SuperData::link<<expression error> >()'
moc_samplefilter.cpp:60:60: error: template argument 1 is invalid
moc_samplefilter.cpp:82:21: error: incomplete type 'QObject' used in nested name specifier
moc_samplefilter.cpp:82:50: error: incomplete type 'QObject' used in nested name specifier
moc_samplefilter.cpp:92:21: error: incomplete type 'QObject' used in nested name specifier
moc_samplefilter.cpp:97:20: error: incomplete type 'QObject' used in nested name specifier

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 adaptors/alsadaptor/alsadaptor.cpp   | 2 +-
 examples/filterplugin/samplefilter.h | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/adaptors/alsadaptor/alsadaptor.cpp b/adaptors/alsadaptor/alsadaptor.cpp
index b0e8f23..b6c3f2b 100644
--- a/adaptors/alsadaptor/alsadaptor.cpp
+++ b/adaptors/alsadaptor/alsadaptor.cpp
@@ -200,7 +200,7 @@ void ALSAdaptor::processSample(int pathId, int fd)
             sensordLogW() << "read(): " << strerror(errno);
             return;
         }
-        QVariant value(buffer);
+        QVariant value {QString(buffer)};
         bool ok;
         double fValue(value.toDouble(&ok));
         if(!ok) {
diff --git a/examples/filterplugin/samplefilter.h b/examples/filterplugin/samplefilter.h
index 43e5016..1ffe930 100644
--- a/examples/filterplugin/samplefilter.h
+++ b/examples/filterplugin/samplefilter.h
@@ -32,6 +32,7 @@
 
 // Include datatypes for input and output.
 #include "timedunsigned.h"
+#include <QObject>
 
 // This is a simplest possible filter, with one input and one output.
 // In case you wish to create more complex filters, with several inputs
